### PR TITLE
Bump kubernetes to v1.28.2

### PR DESCRIPTION
**Pull request by amitsagtani97**
wpb-2324

This pr adds dependencies for deploying kubernetes v1.28.2 by default, with latest kubespray version.

### CODE REVIEW

File: bin/offline-cluster.sh [hunk 1]
```diff
@@ -47,9 +47,7 @@ ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/seed-offline-containerd.yml
 ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/sync_time.yml -v
 
 # Run the rest of kubespray. This should bootstrap a kubernetes cluster successfully:
-ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/kubernetes.yml --skip-tags bootstrap-os,preinstall,container-engine
-
-./bin/fix_default_router.sh
+ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/kubernetes.yml --skip-tags bootstrap-os,preinstall,container-engine,multus
```
Comment (julialongtin) [inline bin/offline-cluster.sh:50]: multius?

File: offline/ci.sh [hunk 2]
```diff
@@ -101,16 +116,16 @@ calling_charts=(
...
 # same as prior.. in most cases.
-wire_calling_version="4.39.0"
+wire_calling_version="4.39.159"
 
 # TODO: Awaiting some fixes in wire-server regarding tagless images
 HELM_HOME=$(mktemp -d)
 export HELM_HOME
 
-helm repo add wire https://s3-eu-west-1.amazonaws.com/public.wire.com/charts
+helm repo add wire https://s3-eu-west-1.amazonaws.com/public.wire.com/charts-develop
```
Comment (julialongtin) [inline offline/ci.sh:128]: aha. this is smart. rewrite this file a bit, so this is the way we do this now?

**Review by julialongtin — APPROVED** [review]

File: bin/offline-cluster.sh (hunk 1 above)
Comment (supersven) [reply bin/offline-cluster.sh:50]: @julialongtin I'm surprised as you are, but there are traces of *multus* in our code base: https://github.com/kubernetes-sigs/kubespray/blob//docs/multus.md?plain=1#L1

Me needs a <scratches-head> emoji.

File: default.nix [hunk 3]
```diff
@@ -26,8 +26,9 @@ rec {
   env = pkgs.buildEnv {
     name = "wire-server-deploy";
     paths = with pkgs; [
-      ansible_2_11
       pythonForAnsible
+      ansible_2_15
```
Comment (supersven) [inline default.nix:30]: Small hint - though it's not very bad: Usually people try to keep the change set (amount of changed lines) small. I.e. I would have just replaced the `1` with a `5`.

But, this is easy to grasp and fine for now.

File: offline/ci.sh [hunk 4]
```diff
@@ -101,16 +116,16 @@ calling_charts=(
 )
 
 # wire_version=$(helm show chart wire/wire-server | yq -r .version)
-wire_version="4.39.0"
+wire_version="4.39.159"
```
Comment (supersven) [inline offline/ci.sh:119]: Hm, so the version that is currently on staging will be the version that our users should install? Was this an active choice by someone? :thinking: 

Or, will this branch not be merged to `master`?

File: bin/offline-cluster.sh [hunk 5]
```diff
@@ -47,9 +47,7 @@ ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/seed-offline-containerd.yml
 ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/sync_time.yml -v
 
 # Run the rest of kubespray. This should bootstrap a kubernetes cluster successfully:
-ansible-playbook -i $INVENTORY_FILE $ANSIBLE_DIR/kubernetes.yml --skip-tags bootstrap-os,preinstall,container-engine
-
-./bin/fix_default_router.sh
```
Comment (supersven) [inline bin/offline-cluster.sh:52]: I cannot find where this script is called now. Am I'm blind (probably) or is `fix_default_router.sh` now unused?

File: nix/pkgs/kubernetes-tools.nix [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 buildGoModule rec {
   pname = "kubernetes";
-  version = "1.23.16";
+  version = "1.28.2";
```
Comment (supersven) [inline nix/pkgs/kubernetes-tools.nix:6]: I'm a bit surprised that the `hash` didn't change. Have you checked if this intented change became active?

On my machine there's still the old version in the nix env:

```
➜  wire-server-deploy git:(bump_kubernetes_to_1_28) kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.16", GitCommit:"60e5135f758b6e43d0523b3277e8d34b4ab3801f", GitTreeState:"archive", BuildDate:"1980-01-01T00:00:00Z", GoVersion:"go1.21.4", Compiler:"gc", Platform:"linux/amd64"}
```

File: nix/pkgs/wire-binaries.nix [hunk 7]
```diff
@@ -8,14 +8,14 @@ let
   # These values are manually kept in sync with:
   # https://github.com/kubernetes-sigs/kubespray/blob/release-2.20/roles/download/defaults/main.yml
```
Comment (supersven) [inline nix/pkgs/wire-binaries.nix:9]: You seem to have updated `kubespray`. Is this comment still fine. Or, does it need to be updated, too?

File: offline/ci.sh (hunk 2 above)
Comment (supersven) [reply offline/ci.sh:128]: The clean solution would be to extract parameters for the parts that depend on the version. And, fill with with what we get from Stefan M.'s `build.json` files: https://github.com/wireapp/wire-builds/blob/108f7eb93bc6d1227f04c23032fab0c58390e5fe/build.json

This refactoring doesn't have to be done now. However, it's good to keep in mind that this changing of versions / Helm repos buried deep inside some files is neither the goal nor the future.

File: nix/pkgs/kubernetes-tools.nix (hunk 6 above)
Comment (amitsagtani97) [reply nix/pkgs/kubernetes-tools.nix:6]: nice find.. I need to update the hash

File: nix/pkgs/wire-binaries.nix (hunk 7 above)
Comment (amitsagtani97) [reply nix/pkgs/wire-binaries.nix:9]: needs to be updated..fixed :)

File: bin/offline-cluster.sh (hunk 5 above)
Comment (amitsagtani97) [reply bin/offline-cluster.sh:52]: this script is no longer needs to be executed, there was some issue with dns configs on k8s v1.23.7 which is no longer found in this newer version.

File: default.nix (hunk 3 above)
Comment (amitsagtani97) [reply default.nix:30]: makes sense, thanks for the suggestion :)
Was unintentional change, I appreciate this suggestion for my long term learning :)

File: offline/ci.sh (hunk 4 above)
Comment (amitsagtani97) [reply offline/ci.sh:119]: there were changes needed in the wire-server helm chart to support k8s 1.28
This change is temporary, goal is to replace it with "4.40" as soon as that is released and than merge

File: bin/offline-cluster.sh (hunk 1 above)
Comment (amitsagtani97) [reply bin/offline-cluster.sh:50]: this multus network plugin playbooks were being run while installing k8s, even the variable to install multus was set to false and was throwing some dependency issues, hence added that tag in the skip list

File: offline/ci.sh (hunk 2 above)
Comment (amitsagtani97) [reply offline/ci.sh:128]: this was temporary change to test the changes needed in helm charts for k8s 1.28
Goal is to use the build.json later.

**Pull request merged**